ospd_openvas/vthelper.py

```diff
@@ -209,10 +210,15 @@ def get_vt_iterator(
             if vt:
                 yield (vt_id, vt)
 
-    def calculate_vts_collection_hash(self) -> str:
+    def calculate_vts_collection_hash(self, path: Path = None) -> str:
```

Comment:
wouldn't it be easier and less error prone to use when the string to calculate it is returned when wanted instead of writing it into a file?

Reply:
the ospd command returns an xml object to the gmv-cli. If you want to work with the string, you have first to process the xml element. E.g. in this case, you can just apply the `sha256sum <file.dump>` to get the hash and compare it.

Reply:
But in this case you make have to make sure that the path exists, is writable, ... and return an appropriate error code if not. So I think that dealing with xml is in comparison easier.